src/models/CommonModel.ts

```diff
@@ -64,14 +160,14 @@ export class CommonModel extends CommonSchema<CommonModel> {
    * @param mergeFrom 
    * @param originalSchema 
    */
-  private static mergeAdditionalProperties(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema) {
+  private static mergeAdditionalProperties(mergeTo: CommonModel, mergeFrom: CommonModel, originalSchema: Schema, alreadyIteratedModels: Map<CommonModel, CommonModel> = new Map()) {
     const mergeToAdditionalProperties = mergeTo.additionalProperties;
     const mergeFromAdditionalProperties = mergeFrom.additionalProperties;
     if (mergeFromAdditionalProperties !== undefined) {
-      if (mergeToAdditionalProperties === undefined) {
-        mergeTo.additionalProperties = mergeFromAdditionalProperties;
+      if (mergeToAdditionalProperties !== undefined) {
+        mergeTo.additionalProperties = CommonModel.mergeCommonModels(mergeToAdditionalProperties, mergeFromAdditionalProperties, originalSchema, alreadyIteratedModels);
       } else {
```

Comment:
I saw above that you are using `return` instead of `else` in cases like here. I think it should be consistent.

Reply:
Made them consistent, thanks 👍